php/libraries/NDB_Config.class.inc

```diff
@@ -234,13 +240,17 @@ class NDB_Config
     /**
      * Gets a setting from the database config tables
      *
-     * @param string  $name The name of the config setting to get
-     * @param integer $id   (optional) The ID of the config setting we're
+     * FIXME This function currently returns a string, array, or null.
+     * Refactoring should be done to better encapsulate the logic in this
+     * function to allow for strong typing.
+     *
+     * @param string   $name The name of the config setting to get
+     * @param int|null $id   (optional) The ID of the config setting we're
      *                      getting to avoid ambiguity.
      *
-     * @return string The value from the database, or null if nothing found.
+     * @return mixed The value from the database, or null if nothing found.
      */
-    function getSettingFromDB($name, $id=null)
+    function getSettingFromDB(string $name, ?int $id = null)
```

Comment:
I can't remember when I added the FIXME above but no, this can return array, string or null

Reply:
the doc block can be updated to indicate a sum type even if the function signature can't, phan might be able to use the slightly tighter bounds to track down a few more bugs from the doc than "mixed".. but I'll merge this as is, since it's relatively minor.